### DISCUSSION TITLE
blog/2023/07/25/conjoint-bayesian-frequentist-guide/

### DISCUSSION THREAD

**Discussion by giscus** [andrewheiss/ath-quarto#87] [Blog comments]
# blog/2023/07/25/conjoint-bayesian-frequentist-guide/

Learn how to use R, {brms}, and {marginaleffects} to analyze conjoint data and find causal and descriptive quantities of interest, both frequentistly and Bayesianly

https://www.andrewheiss.com/blog/2023/07/25/conjoint-bayesian-frequentist-guide/?s=08

**Comment by joscani** [comment]
Nice post. Thanks for sharing

**Comment by yangl-github** [comment]
Nice post and thanks for sharing the source code. I saw you use format: html in your yaml header, instead of output: html_document. Can you please advise the difference between the two options? Some reference reading will be good too. Thanks.

**Reply by andrewheiss** [reply]
That's because I'm using Quarto instead of R Markdown, which [uses different syntax for output formats](https://quarto.org/docs/output-formats/all-formats.html).

**Comment by AlekFisher** [comment]
This is an awesome post, thank you very much! Have you ever run this on partial profile conjoint, where only a subset of attributes are seen on each screen? I use sawtooth for my conjoint analysis but would love to see if I can do most of my analysis in R.

**Comment by Ales-G** [comment]
This is really helpful! thank you very much for writing this post!
I have one question, do you know what is the correct way to formally test if differences in marginal means are statitstically significant? so for example, in this contest, how can I test whether the marginal mean for None group is statistically different from the one for Catholic? 
````
candidate_mms_auto <- cregg::mm(
  candidate,
  selected ~ atmilitary + atreligion + ated +
    atprof + atinc + atrace + atage + atmale,
  id = ~ resID
)
candidate_mms_auto %>% as_tibble()
## # A tibble: 40 × 10
##    outcome  statistic feature          level                  estimate std.error     z         p lower upper
##    <chr>    <chr>     <fct>            <fct>                     <dbl>     <dbl> <dbl>     <dbl> <dbl> <dbl>
##  1 selected mm        Military Service Did Not Serve             0.458   0.00907  50.5 0         0.440 0.475
##  2 selected mm        Military Service Served                    0.543   0.00914  59.4 0         0.525 0.560
##  3 selected mm        Religion         None                      0.556   0.0200   27.8 1.88e-170 0.517 0.595
##  4 selected mm        Religion         Jewish                    0.520   0.0186   28.0 2.77e-172 0.484 0.557
##  5 selected mm        Religion         Catholic                  0.526   0.0177   29.7 2.82e-193 0.491 0.560
##  6 selected mm        Religion         Mainline protestant       0.543   0.0194   28.0 3.93e-172 0.505 0.581
##  7 selected mm        Religion         Evangelical protestant    0.437   0.0200   21.9 4.11e-106 0.398 0.476
##  8 selected mm        Religion         Mormon                    0.417   0.0194   21.5 7.40e-103 0.379 0.455
##  9 selected mm        College          No BA                     0.340   0.0186   18.3 1.12e- 74 0.303 0.376
## 10 selected mm        College          Baptist college           0.482   0.0200   24.1 3.89e-128 0.443 0.521
## # ℹ 30 more rows
````
Thank you in advance for your help on this

**Comment by J0shisIsland** [comment]
Thanks so much for this very helpful post! 

When trying to apply the frequentist marginal means analysis to my own data, I was saddened to see that the cregg package was removed from CRAN entirely (apparently, it hasn't been maintained for the past few years). marginaleffects::marginal_means(), on the other hand, got lost in the recent updates of the package and the output of marginaleffects::avg_predictions() function returns cell means instead of marginal means. 

The simplest alternative I could find was broom.helpers::tidy_marginal_means(), which has an output comparable to that of marginal_means() in your post, except for a different naming convention variable == term and term == value. 

If anyone has thoughts on cregg vs marginaleffects vs broom.helpers in 2026, I'd be curious to hear them!

**Comment by victorhartman** [comment]
Hi Andrew, thank you for this great guide. The explanations on MM and AMCE are really intuitive, which you don't see that often in academic papers.

This guide relies on the 'cregg' package (https://github.com/leeper/cregg) which unfortunately is no receiving support and available on CRAN. There is now also another package available 'projoint' (https://github.com/yhoriuchi/projoint) which seems to work quite well and also corrects for measurement error. 

Would you still advice on using the cregg package? It is still possible to get it working, although it requires some workarounds. For future work, it does seem like projoint is the best option. Would like to hear your thoughts on this!
